web_app/utils/fucntools.py

```diff
@@ -15,3 +21,112 @@ def get_client_ip(request: Request) -> str:
         ip = request.client.host
 
     return ip
+
+
+def download_parquet_file(
+    bucket_name: str | None = GS_BUCKET_NAME, protocol_name: str = None
+) -> None:
+    """
+    Downloads parquet file to local storage from Google Cloud Storage
+    :param bucket_name: Google Cloud Storage bucket name
+    :param protocol_name: Protocol name
+    :return: None
+    """
+    assert protocol_name in [item.value for item in ProtocolIDCodeNames]
+
+    data = dd.read_parquet(
+        GS_BUCKET_URL.format(protocol_name=protocol_name, bucket_name=bucket_name)
+    )
+    dd.to_parquet(df=data, path=f"loans/{protocol_name}_data/")
+
+
+def fetch_user_loans(user_id: str = None, protcol_name: str = None) -> pd.DataFrame:
+    """
+    Fetches user loans data from `.parquet` file
+    :param user_id: User wallet ID
+    :param protcol_name: Protocol name
+    :return: pd.DataFrame
+    """
+    data = pd.read_parquet(
+        path=f"loans/{protcol_name}_data/part.0.parquet",
+    )
+    user = data[data["User"] == user_id]
+    return user.to_dict()
+
+
+def get_user_row_number(user: dict[str, dict[int, str]] = None) -> int:
+    """
+    Returns user row number in `.parquet` file.
+    According to type annotation for `user` parameter,
+    the row stands for `int` python's data type.
+    :param user: dict[str, dict[int, str]]
+    :return: int
+    """
+    return list(user["User"].keys())[0]
+
+
+def get_debt_usd(
+    user_data: dict[str, dict[int, str]] = None, user_row_number: int = None
+) -> float | None:
+    """
+    Returns debt usd value from user_data parameter
+    :param user_data: dict[str, dict[int, str]] = None
+    :param user_row_number: int = None
+    :return: float | None
+    """
+    try:
+        return user_data.get("Debt (USD)", "").get(user_row_number, None)
+
+    except AttributeError:
+        return
+
+
+def get_risk_adjusted_collateral_usd(
+    user_data: dict[str, dict[int, str]] = None, user_row_number: int = None
+) -> float | None:
+    """
+    Returns risk adjusted collateral usd value from user_data parameter
+    :param user_data: dict[str, dict[int, str]] = None
+    :param user_row_number: int = None
+    :return: float | None
+    """
+    try:
+        return user_data.get("Risk-adjusted collateral (USD)", "").get(
```

Comment:
1. Move it `Risk-adjusted collateral (USD)` to constant
2. do it in 2 steps, to cover if first get is `None`:
```
collateral_data = user_data.get("Risk-adjusted collateral (USD)", None) or None
if colletral_data:
   return risk_adjusted_colleral = colleral_data.get(user_row_number) or None
 return None
```